airbyte-scheduler/app/src/main/java/io/airbyte/scheduler/app/JobSubmitter.java

```diff
@@ -114,7 +116,8 @@ void submitJob(final Job job) {
           final Path logFilePath = workerRun.getJobRoot().resolve(LogClientSingleton.LOG_FILENAME);
           final long persistedAttemptId = persistence.createAttempt(job.getId(), logFilePath);
           assertSameIds(attemptNumber, persistedAttemptId);
-          LogClientSingleton.setJobMdc(workerRun.getJobRoot());
+          Configs configs = new EnvConfigs();
```

Comment:
I started exploring that the first time through and ran into weird gotchas and code I wasn't able to conclusively reason about safely.  I'd like to understand this section better, because I think we still have some weird design patterns in play that might not be ideal, but I'm not sure.  Maybe we can pair on this for a bit to investigate.

Reply:
We figured this out with an offline discussion and have a pattern for the refactor that should work fine, using a final local var to pass to the thread.